### PR TITLE
Add GitHub Actions formatter

### LINKED ISSUE #525: Add GitHub Actions Workflow Command output format
Related to https://github.com/crystal-ameba/github-action/issues/19

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions

Would just require outputting in these formats:
```
::notice file={name},line={line},endLine={endLine},title={title}::{message}
::warning file={name},line={line},endLine={endLine},title={title}::{message}
::error file={name},line={line},endLine={endLine},title={title}::{message}
```

### PR DESCRIPTION
**Pull request by Sija**
Closes #525
Refs https://github.com/crystal-ameba/github-action/issues/19

### CODE REVIEW

**Comment by Sija** [comment]
I'm wondering would it be possible to add a [job summary](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#adding-a-job-summary) in a way that wouldn't error out if run outside of GitHub Actions... 🤔

**Review by nobodywasishere — APPROVED** [review]

File: src/ameba/formatter/github_actions_formatter.cr [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    # See for details:
+    # - https://github.com/actions/toolkit/blob/74906bea83a0dbf6aaba2d00b732deb0c3aefd2d/packages/core/src/command.ts#L92-L97
+    # - https://github.com/actions/toolkit/issues/193
+    private def escape_data(string : String) : String
+      string
+        .gsub('%', "%25")
+        .gsub('\r', "%0D")
+        .gsub('\n', "%0A")
+    end
```
Comment (veelenga) [inline src/ameba/formatter/github_actions_formatter.cr:58]: shouldn't we escape a semicolor here.  Example:

```
$ cat actions-formatter.cr                                                                                                                                                             
begin
rescue Exception
rescue Foo::Exception
end

$ ./bin/ameba actions-formatter.cr -f github-actions                                                                                                                                     
::warning file=actions-formatter.cr,line=3,col=8,endLine=3,endColumn=21,title=Lint/ShadowedException::Shadowed exception found: Foo::Exception
```

Comment (Sija) [reply src/ameba/formatter/github_actions_formatter.cr:58]: Nope. See the linked sources for the reference.

Comment (Sija) [reply src/ameba/formatter/github_actions_formatter.cr:58]: Looks like it's working:

image

**Review by veelenga — APPROVED** [review]
lgtm

**Pull request merged**